frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts

```diff
@@ -1,10 +1,33 @@
 import { K8sResourceKind } from '@console/internal/module/k8s';
 import { NO_PROVISIONER, OCS_INTERNAL_CR_NAME, CEPH_STORAGE_NAMESPACE } from '../../constants';
 
+export const createDeviceSet = (scName: string, osdSize: string, portable: boolean): DeviceSet => {
```

Comment:
`scName` and `storage` are optional param in `getOCSRequestData`, here also it should be optional too. 
`getOCSRequestData` without these parameters are used at some places for just getting few details.

Reply:
You want me to create device set without storage class ? That is not possible.
I have fixed the places where it was getting used.
See here https://github.com/openshift/console/pull/6028#discussion_r457850445

Reply:
No, I am not saying that. I was referring to the way `getOCSRequestData` method has been used at some places(in order to use the same code), without passing the storage class, just to fetch few values.
It has been used at other places too, where passing storage class name is not required -
1. https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/components/ocs-install/create-form.tsx#L102
2. https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx#L98

For `getOCSRequestData` -> `scName` and `storage` are optional params, and `createDeviceSet` is expecting these values. So, `createDeviceSet` should be called only when these params are present. Can you add the check atleast?

Reply:
I am aware and I must say that its not nice to have request data to be used for fetching name when we have a constant.
I planned to do in followup refactors but I did the change now.
The change is not to check for sc but only use ot for creating storage cluster.
Also `getOCSRequestData` -> `createInternalAndAttachedStorageCluster`
`createInternalAndAttachedStorageCluster` is using internally the cluster name as `OCS_INTERNAL_CR_NAME`, so using one constant at all requirements of storage cluster name is right.

Reply:
`getOCSRequestData` was used just not for the name, it was used for getting other values too like replica etc. Hence, this approach was used. 
It should be named as `getOCSRequestData` because here we are fetching the request data and not creating a storage cluster. You can replace `OCS` word, if you want but naming this like `createInternalAndAttachedStorageCluster` doesn't seem right.

Reply:
@gnehapk I request you to review my PR, I have added replica constant.
this function will be under makeOCSRequest.

https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/components/ocs-install/create-form.tsx#L60-L67